core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java

```diff
@@ -82,8 +84,11 @@ public void delete(PageId pageId) throws IOException, PageNotFoundException {
     }
     Files.delete(p);
     mSize.decrementAndGet();
-    if (!Files.newDirectoryStream(p.getParent()).iterator().hasNext()) {
-      Files.delete(p.getParent());
+    Path parent = Preconditions.checkNotNull(p.getParent(), "parent");
```

Comment:
please make the error message more explicit.

Reply:
Done.